pkg/types/interfacetests/utils.go

```diff
@@ -36,12 +41,88 @@ func runTests[T TestingT[T]](t T, tester BasicTester[T], tests []testcase[T]) {
 	}
 }
 
+func batchChainWrite[T TestingT[T]](t T, tester ChainComponentsInterfaceTester[T], batchCallEntry BatchCallEntry, mockRun bool) {
+	if mockRun {
+		cw := tester.GetChainWriter(t)
+		err := cw.SubmitTransaction(tests.Context(t), AnyContractName, "batchChainWrite", batchCallEntry, "", "", nil, big.NewInt(0))
+		require.NoError(t, err)
+		return
+	}
+	nameToAddress := make(map[string]string)
+	boundContracts := tester.GetBindings(t)
+	for _, bc := range boundContracts {
+		nameToAddress[bc.Name] = bc.Address
+	}
+
+	for contractName, contractBatch := range batchCallEntry {
+		require.Contains(t, nameToAddress, contractName)
+		for _, readEntry := range contractBatch {
+			val, isOk := readEntry.ReturnValue.(*TestStruct)
+			if !isOk {
+				require.Fail(t, "expected *TestStruct for contract: %s read: %s, but received %T", contractName, readEntry.Name, readEntry.ReturnValue)
+			}
+			SubmitTransactionToCW(t, tester, "addTestStruct", val, types.BoundContract{Name: contractName, Address: nameToAddress[contractName]}, types.Unconfirmed)
+		}
+	}
+}
+
+func SubmitTransactionToCW[T TestingT[T]](t T, tester ChainComponentsInterfaceTester[T], method string, args any, contract types.BoundContract, status types.TransactionStatus) string {
+	tester.DirtyContracts()
+	txID := uuid.New().String()
+	cw := tester.GetChainWriter(t)
+	err := cw.SubmitTransaction(tests.Context(t), contract.Name, method, args, txID, contract.Address, nil, big.NewInt(0))
+	require.NoError(t, err)
+
+	err = WaitForTransactionStatus(t, tester, txID, status, false)
+	require.NoError(t, err)
+
+	return txID
+}
+
+func WaitForTransactionStatus[T TestingT[T]](t T, tester ChainComponentsInterfaceTester[T], txID string, status types.TransactionStatus, mockRun bool) error {
+	ctx, cancel := context.WithTimeout(tests.Context(t), 15*time.Minute)
+	defer cancel()
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	fmt.Printf("Waiting for transaction %s to reach status %d\n", txID, status)
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("transaction %s not finalized within timeout period", txID)
+		case <-ticker.C:
+			if mockRun {
+				tester.GenerateBlocksTillConfidenceLevel(t, "", "", primitives.Finalized)
+				return nil
+			}
+			current, err := tester.GetChainWriter(t).GetTransactionStatus(ctx, txID)
+			if err != nil {
+				return fmt.Errorf("failed to get transaction status: %w", err)
+			}
+
+			if current == types.Failed || current == types.Fatal {
+				return fmt.Errorf("transaction %s has failed or is fatal", txID)
+			} else if current >= status {
+				fmt.Printf("Transaction %s reached status: %d\n", txID, current)
```

Comment:
please remove

Reply:
I assume you mean just the Printf statement, just removed.